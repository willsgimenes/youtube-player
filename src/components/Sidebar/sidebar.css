.tab-nav {
  padding: 20px;
  width: 90%;
  margin: 0 auto;
}

.tab-nav ul {
  padding: 0;
  margin: 0;
  top: 0;
  height: auto;
  width: 100%;
  display: flex;
  flex-wrap: nowrap;
  flex-flow:row wrap;
  box-sizing:border-box;
}

.tab-nav ul li {
  flex:1;
  box-sizing:border-box;
  border:1px solid #167efb;
  border-right:none;
  text-align: center;
}

.tab-nav ul li a {
  padding: 10px 5px;
  display: block;
  text-decoration: none;
}

.tab-nav ul li:first-child {
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  border-right:0;
}

.tab-nav ul li:last-child{
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  border-right:1px solid #167efb;
}

.tab-nav__selected {
  background:  #167efb;
  color: #fff;
}
