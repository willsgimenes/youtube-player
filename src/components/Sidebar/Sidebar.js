import React from 'react'
import { NavLink } from 'react-router-dom'
import './sidebar.css'

class Sidebar extends React.Component {
  render () {
    return (
      <div className='tab-nav'>
        <ul>
          <li><NavLink to='/playlist' activeClassName={'tab-nav__selected'}>Playlist</NavLink></li>
          <li><NavLink to='/search' activeClassName={'tab-nav__selected'}>Search</NavLink></li>
        </ul>
      </div>
    )
  }
}

export { Sidebar }
