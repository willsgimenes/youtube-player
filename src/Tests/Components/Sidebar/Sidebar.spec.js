/* global describe */
/* global it */
/* global expect */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Sidebar } from '../../../components/Sidebar/Sidebar'

Enzyme.configure({ adapter: new Adapter() })

describe('List component', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Sidebar />)
    expect(wrapper.find('.tab-nav')).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })
})

it('should have only two links', () => {
  const wrapper = shallow(<Sidebar />)
  expect(wrapper.find('ul li')).toHaveLength(2)
  expect(wrapper).toMatchSnapshot()
})
