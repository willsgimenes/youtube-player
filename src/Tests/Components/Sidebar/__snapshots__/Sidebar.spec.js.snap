// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/playlist"
                >
                        Playlist
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/search"
                >
                        Search
                </NavLink>
        </li>
</ul>,
      "className": "tab-nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/playlist"
            >
                        Playlist
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/search"
            >
                        Search
            </NavLink>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="tab-nav__selected"
              aria-current="page"
              to="/playlist"
>
              Playlist
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "tab-nav__selected",
              "aria-current": "page",
              "children": "Playlist",
              "to": "/playlist",
            },
            "ref": null,
            "rendered": "Playlist",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="tab-nav__selected"
              aria-current="page"
              to="/search"
>
              Search
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "tab-nav__selected",
              "aria-current": "page",
              "children": "Search",
              "to": "/search",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    <NavLink
                              activeClassName="tab-nav__selected"
                              aria-current="page"
                              to="/playlist"
                    >
                              Playlist
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="tab-nav__selected"
                              aria-current="page"
                              to="/search"
                    >
                              Search
                    </NavLink>
          </li>
</ul>,
        "className": "tab-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="tab-nav__selected"
                            aria-current="page"
                            to="/playlist"
              >
                            Playlist
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="tab-nav__selected"
                            aria-current="page"
                            to="/search"
              >
                            Search
              </NavLink>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="tab-nav__selected"
                aria-current="page"
                to="/playlist"
>
                Playlist
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "tab-nav__selected",
                "aria-current": "page",
                "children": "Playlist",
                "to": "/playlist",
              },
              "ref": null,
              "rendered": "Playlist",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="tab-nav__selected"
                aria-current="page"
                to="/search"
>
                Search
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "tab-nav__selected",
                "aria-current": "page",
                "children": "Search",
                "to": "/search",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`should have only two links 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/playlist"
                >
                        Playlist
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/search"
                >
                        Search
                </NavLink>
        </li>
</ul>,
      "className": "tab-nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/playlist"
            >
                        Playlist
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="tab-nav__selected"
                        aria-current="page"
                        to="/search"
            >
                        Search
            </NavLink>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="tab-nav__selected"
              aria-current="page"
              to="/playlist"
>
              Playlist
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "tab-nav__selected",
              "aria-current": "page",
              "children": "Playlist",
              "to": "/playlist",
            },
            "ref": null,
            "rendered": "Playlist",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="tab-nav__selected"
              aria-current="page"
              to="/search"
>
              Search
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "tab-nav__selected",
              "aria-current": "page",
              "children": "Search",
              "to": "/search",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    <NavLink
                              activeClassName="tab-nav__selected"
                              aria-current="page"
                              to="/playlist"
                    >
                              Playlist
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="tab-nav__selected"
                              aria-current="page"
                              to="/search"
                    >
                              Search
                    </NavLink>
          </li>
</ul>,
        "className": "tab-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="tab-nav__selected"
                            aria-current="page"
                            to="/playlist"
              >
                            Playlist
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="tab-nav__selected"
                            aria-current="page"
                            to="/search"
              >
                            Search
              </NavLink>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="tab-nav__selected"
                aria-current="page"
                to="/playlist"
>
                Playlist
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "tab-nav__selected",
                "aria-current": "page",
                "children": "Playlist",
                "to": "/playlist",
              },
              "ref": null,
              "rendered": "Playlist",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="tab-nav__selected"
                aria-current="page"
                to="/search"
>
                Search
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "tab-nav__selected",
                "aria-current": "page",
                "children": "Search",
                "to": "/search",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
